### ISSUE TITLE
[Bug]: Component definition is missing display name when using React `forwardRef`.

### ISSUE THREAD

**Issue by devjiwonchoi** [jsx-eslint/eslint-plugin-react#3590]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

Lint for React `forwardRef` gives an error `Component definition is missing display name`.

Code to reproduce:

```js
import { forwardRef } from 'react'

const test = forwardRef(() => {
})
```

### Expected Behavior

Should not give any errors.

Can be resolved by `as` alias, or `Test.displayName = 'Test';`

### Additional Information

Failing on test cases also.

![Screenshot 2023-06-26 at 6 55 23 PM](https://github.com/jsx-eslint/eslint-plugin-react/assets/120007119/177d846c-5325-432d-ac5c-e50dcafbaebf)

Passes fail case test.

![Screenshot 2023-06-26 at 6 57 14 PM](https://github.com/jsx-eslint/eslint-plugin-react/assets/120007119/572c4964-7cca-4cd9-a557-7f41f9c7c21f)

![Screenshot 2023-06-26 at 6 57 30 PM](https://github.com/jsx-eslint/eslint-plugin-react/assets/120007119/43403aa5-b37c-4a2c-b59a-89d6545d2f12)

### eslint-plugin-react version

v7.32.2

### eslint version

v8.36.0

### node version

v18.16.1

**Comment by devjiwonchoi** [comment]
Found out it is intended, closing Issue.

![Screenshot 2023-06-26 at 7 27 44 PM](https://github.com/jsx-eslint/eslint-plugin-react/assets/120007119/33433c72-707d-4758-8b9d-fd33e446b56c)


## Reference
- [Disallow missing displayName in a React component definition](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/display-name.md#disallow-missing-displayname-in-a-react-component-definition-reactdisplay-name)
